src/common/snippets/src/lowered/pass/propagate_subtensors.cpp

```diff
@@ -15,14 +15,43 @@ namespace snippets {
 namespace lowered {
 namespace pass {
 namespace {
+
+// The algorithm uses the following special values in subtensors/shapes:
+// 1. Dynamic value in subtensor/shape : SIZE_MAX
+// 2. Full dimension in subtensor      : SIZE_MAX - 1
+// 3. Default value of `new_dim_value` : SIZE_MAX - 2
+// 4. `Forced` special dynamic value   : SIZE_MAX - 3
+//
+// We have to introduce `FORCED_DYNAMIC_VALUE` to distinguish `new_dim_value = DYNAMIC`
+// from the real dynamic values in subtensors and shapes and force this value in subtensors.
+// For example, there is Brgemm with the following info in the tail Loop:
+// Input 0: shape [?, ?], existing subtensor [32, FULL_DIM]
+// Input 1: shape [?, ?], existing subtensor [FULL_DIM, FULL_DIM]
+// Output : shape [?, ?], existing subtensor [32, FULL_DIM]
+// If the user wants to force `?` in the place of `32` in subtensors, the steps will be:
+// 1. Set `?` to subtensor and shape of Input 0 :
+//    shape [?, ?] (shape has not been changed!), new subtensor [?, FULL_DIM]
+// 2. Make shape inference of Brgemm and get Output:
+//    shape [?, ?] (shape has not been changed!), existing subtensor [FULL_DIM, FULL_DIM]
+// 3. Update subtensor on output using shape:
+//    new_subtensor[i] = std::min(planar_shape[i], subtensor[i]); // i = 0: std::min(SIZE_MAX(?), 32)
+//    new subtensor [32, FULL_DIM] - has not been changed! But should be [?, FULL_DIM]
+// Conclusion: we have to distinguish forced dynamic value with existing dynamic values in shape and subtensor
```

Comment:
Can't agree, let's discuss offline.
I think that the problem is in shape inference rules:
If `[32, FULL_DIM] x [ FULL_DIM, FULL_DIM] -->  [32,  FULL_DIM]` works then
    `[?, FULL_DIM] x [ FULL_DIM, FULL_DIM] -->  [?,  FULL_DIM]` should also work without additional `forced` values.

Reply:
Let's discuss offline.